File: go/45.go
func jump(nums []int) int {
	var start = 0
	var end = 0
	var l = len(nums)
	var ans = 0
	for end < l-1 {
		ans++
		tmp := end
		for i := start; i <= end; i++ {
			if i+nums[i] > tmp {
				tmp = i + nums[i]
				start = i
			}
		}
		end = tmp

	}
	return ans
}

File: java/45.java
class Solution {
    public int jump(int[] nums) {
        int l = nums.length;
        int start = 0,end = 0;
        int ans = 0;
        while(end < l-1){
            ans ++;
            int tmp_end = end;
            for(int i = start; i <=end;i++){
                if(i + nums[i] > tmp_end){
                    tmp_end = i + nums[i];
                    start = i;
                }
            }
            // start = tmp_start;
            end = tmp_end;
        }
        return ans;
        
    }
}